sdk/workloads/Azure.ResourceManager.Workloads/tests/Tests/SviCRUDTests.cs

```diff
@@ -0,0 +1,216 @@
+            await sapCentralServerInstanceResource.AddTagAsync("TestKey1", "TestValue1");
+            await sapDatabaseInstanceResource.AddTagAsync("TestKey1", "TestValue1");
+
+            Thread.Sleep(GetStatusIntervalinMillis);
+
+            sapVirtualInstance =
+                await Client.GetSapVirtualInstanceResource(sviResourceID).GetAsync();
+            sapDatabaseInstanceResource =
+                await Client.GetSapDatabaseInstanceResource(dbServerResourceID).GetAsync();
+            sapApplicationServerInstanceResource =
+                await Client.GetSapApplicationServerInstanceResource(appServerResourceID).GetAsync();
+            sapCentralServerInstanceResource =
+                await Client.GetSapCentralServerInstanceResource(csServerResourceID).GetAsync();
+
+            var firsttag = sapVirtualInstance.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+
+            firsttag = sapDatabaseInstanceResource.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+            await sapDatabaseInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+
+            firsttag = sapCentralServerInstanceResource.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+            await sapCentralServerInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+
+            firsttag = sapApplicationServerInstanceResource.Data.Tags.Values.GetEnumerator();
+            firsttag.MoveNext();
+            Assert.AreEqual(firsttag.Current, "TestValue1");
+            await sapApplicationServerInstanceResource.GetTagResource().DeleteAsync(WaitUntil.Completed);
+        }
+
+        /// <summary>
+        /// Gets an object of SAP Workload Resource for Single Server that should be used for PUT call.
+        /// </summary>
+        /// <param name="infraRgName"> Contains the infra RG
+        /// used as app resource group. </param>
+        /// <returns>PHP Workload Resource.</returns>
+        private SapVirtualInstanceData GetSingleServerPayloadToPut(string infraRgName, bool isInstall)
+        {
+            // Reading constants from files.
+            string sshPublicKey = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "SshKeyPublic.txt"));
+            string fileName = isInstall ? "SingleServerInstall.json" : "SingleServerCreate.json";
+
+            var testSapWorkloadJson = File.ReadAllText(Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), fileName));
+
+            // Updating payload at runtme.
+            dynamic dynamicObject = JsonConvert.DeserializeObject<dynamic>(testSapWorkloadJson);
```

Comment:
I don't think it's the best practice to simultaneously use Newtonsoft.json and System.Text.Json. Can you change it?
```suggestion
            var inputData = SapVirtualInstanceData.DeserializeSapVirtualInstanceData(JsonDocument.Parse(testSapWorkloadJson).RootElement);
```
After that, you can cast the base type to its derived type to set the value that you want like:
```csharp
((DeploymentConfiguration)testObject.Configuration).InfrastructureConfiguration.AppResourceGroup = infraRgName;
```

Reply:
Done